beacon/blsync/engineclient.go

```diff
@@ -47,7 +47,10 @@ func startEngineClient(config *lightClientConfig, rpc *rpc.Client, headCh <-chan
 		cancelRoot: cancel,
 	}
 	ec.wg.Add(1)
-	go ec.updateLoop(headCh)
+	go func() {
+		defer ec.wg.Done()
```

Comment:
I don't really see the difference between two approaches. They should be equivalent, isn't it?

Reply:
@rjl493456442 Yes sir, in most cases they are equivalent.

But keeping `wg.Done` close to `wg.Add` is clearer for us to ensure we call `wg.Done` correctly. 

And if we just wanna run `ec.updateLoop` in the main process for a test, wrapping `wg.Done` in `ec.updateLoop` will force us to call `wg.Add` even though I don't wanna run in a goroutine.

IMO letting the caller decide the way to call `ec.updateLoop` is more flexible.

Reply:
I think so too, I think if we follow the previous behavior, we must run this function in a goroutine